### PR TITLE
add findByIdSync API

### PR DESCRIPTION
**Pull request by staltz**
- [x] Pending on https://github.com/ssb-ngi-pointer/ssb-db2/pull/260

Adds `findByIdSync`, similar to `findById`, but synchronous (see e.g. similar to `fs.readFile` and `fs.readFileSync`). The synchronous variant requires calling `loadState(cb)` before usage, otherwise it throws an error.

I also refactored/moved stuff from query.js to feeds-lookup.js.

### CODE REVIEW

File: feeds-lookup.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    for (const key of keys) {
+      metadata[key] = content[key]
+    }
+    details.metadata = metadata
+    return details
+  }
+
+  function updateLookup(msg) {
+    const content = msg.value.content
+    if (content.type.startsWith('metafeed/add/')) {
```
Comment (arj03) [inline feeds-lookup.js:95]: This needs a validate.isValid here

**Comment by arj03** [comment]
Found one thing that needs fixing before merging. Otherwise look good. It's great to see that this can use the general authorIsBendyButtV1. Should be plenty fast for EBT. Nice work.

**Comment by staltz** [comment]
Thanks for taking a look. How about commit https://github.com/ssb-ngi-pointer/ssb-meta-feeds/pull/42/commits/5e76cef3f164894afb12aa96f96a0fa578763d11 ?

**Review by arj03 — APPROVED** [review]
Yup, that should work :)

**Pull request merged**